test/functional/tools/discover_sort_by.xml

```diff
@@ -0,0 +1,26 @@
+<tool id="discover_sort_by" name="discover_sort_by" version="0.1.0">
+  <command>
+    mkdir outputs; cd outputs; awk '{ print \$2 > (\$1 ".tabular") }' $input1
+  </command>
+  <inputs>
+    <param name="input1" type="data" label="Input Table" help="Table to split on first column" format="tabular" />
+  </inputs>
+  <outputs>
+    <collection name="collection_numeric_name" type="list" label="Table split on first column">
+      <discover_datasets pattern="__name_and_ext__" sort_by="numeric_name" directory="outputs" />
+    </collection>
+    <collection name="collection_rev_numeric_name" type="list" label="Table split on first column">
+      <discover_datasets pattern="__name_and_ext__" sort_by="reverse_numeric_name" directory="outputs" />
+    </collection>
+    <data name="data_numeric_name">
+      <discover_datasets pattern="__name_and_ext__" format="tabular" assign_primary_output="true" sort_by="reverse_numeric_name" directory="outputs" visible="true"/>
+    </data>
+  </outputs>
+  <tests>
+    <test>
+      <param name="input1" value="tinywga.fam" />
+      <output_collection name="collection_numeric_name" type="list" count="11"/>
```

Comment:
Don't you want to ascertain the new sort order ? You can list `<element/>` items here

Reply:
That's true, but I did not expect the order of the elements is relevant. Lets try.

Reply:
Test currently do not check for the order, but only if the elements contained in the test definition are produced by the test: https://github.com/galaxyproject/galaxy/blob/94390c2af8fb1ce8f54c6665810991da36e6331a/lib/galaxy/tool_util/verify/interactor.py#L717

This would be (quite) easy to change, but 

- the test definition is a dict and dictionary order can only be used for python >= 3.7 
- dictionary order would always be lexicographic as defined for strings
- it seems that the dictionary order is lost in the `get_tool_tests` function called here https://github.com/galaxyproject/galaxy/blob/94390c2af8fb1ce8f54c6665810991da36e6331a/lib/galaxy/tool_util/verify/interactor.py#L803

Seems difficult. 

Spontaneous ideas

- sort the test elements in verify_elements as given by `sort_by`
- leave it as is and consider the test just checking for bugs

Reply:
> * the test definition is a dict and dictionary order can only be used for python >= 3.7

3.6, it was made official in 3.7



> * dictionary order would always be lexicographic as defined for strings

this should be by insertion order on newer python's, right ?

Anyway, I agree this is not as straightforward. Maybe add a <!--TODO: test actual order --> in the test, and then we can reconsider once sort order is guaranteed or we decide to write an API test, which could also assert this.

Reply:
> 3.6, it was made official in 3.7

true

> this should be by insertion order on newer python's, right ?

also true, my mistake

> Anyway, I agree this is not as straightforward. Maybe add a in the test, and then we can reconsider once sort order is guaranteed or we decide to write an API test, which could also assert this.

Still I have the urge to understand this :) Somewehere here:

https://github.com/galaxyproject/galaxy/blob/94390c2af8fb1ce8f54c6665810991da36e6331a/lib/galaxy/tool_util/verify/interactor.py#L141

the order seems to get lost. Any suggestions where I could dig?